jax/_src/dispatch.py

```diff
@@ -545,7 +529,7 @@ def _input_handler(backend: Backend,
   in_avals, which_explicit = util.unzip2(in_type)
   # Check whether we actually need an input_handler.
   needs_implicit = which_explicit and not all(which_explicit)
-  needs_out_handling = any(type(d) is core.InDBIdx for a in out_type or []
+  needs_out_handling = any(type(d) is core.InDBIdx for a, _ in out_type or []
```

Comment:
Sometimes variables named `out_type` are of type `Tuple[Tuple[AbstractValue, bool], ...]` (aka `core.OutputType`) and sometimes they are of type `Sequence[AbstractValue]` (internally in this file). Either the type of this variable changed or this was a bug before (which would've caused `needs_out_handling` to always be False). Either way, it's consistent/correct now!